multi/imxrt-multi/spi.c

```diff
@@ -0,0 +1,441 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT SPI driver
+ *
+ * Copyright 2019 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <sys/interrupt.h>
+#include <sys/threads.h>
+#include <sys/pwman.h>
+#include <sys/msg.h>
+#include <unistd.h>
+#include <errno.h>
+
+
+#include "imxrt-multi.h"
+#include "common.h"
+#include "config.h"
+#include "spi.h"
+
+#define SPI1_POS 0
+#define SPI2_POS (SPI1_POS + SPI1)
+#define SPI3_POS (SPI2_POS + SPI2)
+#define SPI4_POS (SPI3_POS + SPI3)
+
+#define SPI_CNT (SPI1 + SPI2 + SPI3 + SPI4)
+
+#define MAX_FRAME_SZ 0x1000
+#define WORD_SIZE sizeof(uint32_t)
+#define MAX_FIFOSZ_BYTES 16 * WORD_SIZE
+
+#define DESERIALIZE_WORD(buff) \
+	buff[3] | (buff[2] << 8) | (buff[1] << 16) | (buff[0] << 24)
+
+#define SERIALIZE_WORD(buff, word, start)		\
+	for (int i = start - 1; i >= 0; --i) 		\
+		*buff++ = (word >> (i * 8)) & 0xff;
```

Comment:
Warning! You are modifying buff pointer implicitly. Makes it easy to produce a magnificent bug ;)

Reply:
If you make it function instead of macro (which I suggest), it would be explicite, that this pointer is modified (as you would pass non const **buff to this function)